server/events/vcs/gitlab_client.go

```diff
@@ -185,6 +189,61 @@ func (g *GitlabClient) ReactToComment(repo models.Repo, commentID int64, reactio
 }
 
 func (g *GitlabClient) HidePrevCommandComments(repo models.Repo, pullNum int, command string) error {
+	var allComments []*gitlab.Note
+
+	nextPage := 0
+	for {
+		g.logger.Debug("/projects/%v/merge_requests/%d/notes", repo.FullName, pullNum)
+		comments, resp, err := g.Client.Notes.ListMergeRequestNotes(repo.FullName, pullNum,
+			&gitlab.ListMergeRequestNotesOptions{
+				Sort:        gitlab.String("asc"),
+				OrderBy:     gitlab.String("created_at"),
+				ListOptions: gitlab.ListOptions{Page: nextPage},
+			})
+		if err != nil {
+			return errors.Wrap(err, "listing comments")
+		}
+		allComments = append(allComments, comments...)
+		if resp.NextPage == 0 {
+			break
+		}
+		nextPage = resp.NextPage
+	}
+
+	currentUser, _, err := g.Client.Users.CurrentUser()
+	if err != nil {
+		return errors.Wrap(err, "error getting currentuser")
+	}
+
+	summaryHeader := fmt.Sprintf("<!--- +-Superseded Command-+ ---><details><summary>Superseded Atlantis %s</summary>", command)
+	summaryFooter := "</details>"
+	lineFeed := "\n"
+
+	for _, comment := range allComments {
+		// Only process non-system comments authored by the Atlantis user
+		if comment.System || (comment.Author.Username != "" && !strings.EqualFold(comment.Author.Username, currentUser.Username)) {
+			continue
+		}
+
+		body := strings.Split(comment.Body, "\n")
+		if len(body) == 0 {
+			continue
+		}
+		firstLine := strings.ToLower(body[0])
+		// Skip processing comments that don't contain the command or contain the summary header in the first line
+		if !strings.Contains(firstLine, strings.ToLower(command)) || firstLine == strings.ToLower(summaryHeader) {
+			continue
+		}
+
+		g.logger.Debug("Updating merge request note: Repo: '%s', MR: '%d', comment ID: '%d'", repo.FullName, pullNum, comment.ID)
+		supersededComment := summaryHeader + lineFeed + comment.Body + lineFeed + summaryFooter + lineFeed
+
+		if _, _, err := g.Client.Notes.UpdateMergeRequestNote(repo.FullName, pullNum, comment.ID,
```

Comment:
Will any of these g.Client() calls result in more permissions needed for atlantis within gitlab ? If so, any additional docs that need to be added?

Reply:
No, the `api` token scope is all that is needed which is already documented.